### PR TITLE
feat(//core/plugins): Add adaptive_max_pool2d plugin, enable the plugins to run on GPU

### PR DESCRIPTION
**Pull request by peri044**
# Description

Add adaptive max pool2d plugin to the new plugins interface. Add changes in the PR https://github.com/NVIDIA/TRTorch/pull/328 to the latest codebase. 

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/converters/impl/pooling.cpp [hunk 1]
```diff
@@ -60,77 +60,53 @@ bool AdaptivePoolingConverter(
...
+  fc.nbFields = f.size();
+  fc.fields = f.data();
+  /*====== PLUGIN PARAMETERS CONFIGURATION COMPLETED ======*/
 
-    std::vector<int64_t> window(out_size.nbDims);
-    for (int64_t i = 0; i < out_size.nbDims; i++) {
-      window[window.size() - 1 - i] =
-          in_shape[in_shape.size() - 1 - i] - (out_size.d[out_size.nbDims - 1 - i] - 1) * stride[stride.size() - 1 - i];
-    }
+  LOG_WARNING(
```
Comment (narendasan) [inline core/conversion/converters/impl/pooling.cpp:102]: This warning is unclear, seems like some text is missing from the original

**Review by narendasan — COMMENTED** [review]
Minor comment but seems fine in general

File: core/conversion/converters/impl/pooling.cpp (hunk 1 above)
Comment (narendasan) [reply core/conversion/converters/impl/pooling.cpp:102]: Also static shape would not trigger adaptive pooling

Comment (peri044) [reply core/conversion/converters/impl/pooling.cpp:102]: Fixed it now.

Comment (peri044) [reply core/conversion/converters/impl/pooling.cpp:102]: Fixed it now

Comment (peri044) [reply core/conversion/converters/impl/pooling.cpp:102]: Removed the static shape handling via TensorRT. That was the reason for adaptive avg pool bugs as listed in #328 . Adaptive pool (max/avg) regardless of static/dynamic shapes run via aten library now.

Comment (narendasan) [reply core/conversion/converters/impl/pooling.cpp:102]: Is there a reason for this? We should be able to calculate the parameters if we know the input shape

Comment (peri044) [reply core/conversion/converters/impl/pooling.cpp:102]: No reason. I followed the PR and replaced it but I can look into calculating the parameters. In the PR, the tests were failing for some shapes with the static shape code.

**Pull request merged**